### DISCUSSION TITLE
Inquiry about the retention policy

### DISCUSSION THREAD

**Discussion by hyeyoung-leee** [goharbor/harbor#20688] [Q&A] [answered]
Hi,
I have a question about the retention policy.

I have two artifacts with push time of `7/3/24, 3:34 PM` and `3/4/24, 4:47 PM`, respectively.
<img width="1643" alt="스크린샷 2024-07-03 오후 4 28 52" src="https://github.com/goharbor/harbor/assets/120620071/ee673e27-3e5e-4516-8500-f0595f0a18f2">

When I dry run the retention policies on `7/3/24, 4:34 PM`, both artifacts are retained.
For the repositories matching **, retain the `artifacts pushed within the last 5 days` with tags matching ** with untagged
For the repositories matching **, retain the `most recently pushed 5 artifacts` with tags matching ** with untagged

Is this correct?
I understand that the retention policy is applied using an "or" logic, so shouldn't the image with the timestamp `3/4/24, 4:47 PM` be deleted?

**Comment by chlins** [comment, marked as answer]
Yes, the logic between retention rules is 'OR' which means the artifact will be retained if it matched any rule, `3/4/24, 4:47 PM` the artifact matches `most recently pushed 5 artifacts` because you only have 2(less than 5) artifacts  in this repo.
